processor/src/main/resources/org/mapstruct/ap/internal/model/StreamMappingMethod.ftl

```diff
@@ -180,5 +180,5 @@
     </@compress>
 </#macro>
 <#macro returnLocalVarDefOrUpdate>
-    <#if canReturnImmediatelly><#if returnType.name != "void">return </#if><#elseif !needVarDefine><@iterableLocalVarDef/> ${resultName} = <#else>${resultName} = </#if><#nested />
+    <#if canReturnImmediatelly><#if returnType.name != "void">return </#if><#elseif needVarDefine><@iterableLocalVarDef/> ${resultName} = <#else>${resultName} = </#if><#nested />
```

Comment:
ah.. I tried the stuff above, but apparently missed this one.. In my case, all kinds of stuff started to fail in the iterable-iterable mappings...